--- a/ranger/container/settings.py
+++ b/ranger/container/settings.py
@@ -280,7 +280,7 @@
         if path:
             if path not in self._localsettings:
                 try:
-                    regex = re.compile(re.escape(path))
+                    regex = re.compile(path)
                 except re.error:  # Bad regular expression
                     return
                 self._localregexes[path] = regex
