--- a/ranger/core/fm.py
+++ b/ranger/core/fm.py
@@ -74,9 +74,11 @@
         self.hostname = socket.gethostname()
         self.home_path = os.path.expanduser('~')

-        mimetypes.knownfiles.append(os.path.expanduser('~/.mime.types'))
+        mimetypes.knownfiles.append(os.path.expanduser('~/.local/share/ranger/mime.types'))
         mimetypes.knownfiles.append(self.relpath('data/mime.types'))
-        self.mimetypes = mimetypes.MimeTypes()
+
+        filenames = [f for f in mimetypes.knownfiles if os.path.isfile(f)]
+        self.mimetypes = mimetypes.MimeTypes(filenames=filenames)

     def initialize(self):
         """If ui/bookmarks are None, they will be initialized here."""
