diff --git i/ranger/core/fm.py w/ranger/core/fm.py
index 6a7fa3a0..902a02b1 100644
--- i/ranger/core/fm.py
+++ w/ranger/core/fm.py
@@ -75,7 +75,7 @@ class FM(Actions,  # pylint: disable=too-many-instance-attributes
         self.home_path = os.path.expanduser('~')
 
-        mimetypes.knownfiles.append(os.path.expanduser('~/.mime.types'))
+        mimetypes.knownfiles.append(os.path.expanduser('~/.local/share/ranger/mime.types'))
         mimetypes.knownfiles.append(self.relpath('data/mime.types'))
         self.mimetypes = mimetypes.MimeTypes()
 
